pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/codestyle/UnnecessaryCastRule.java

```diff
@@ -107,6 +108,13 @@ public Object visit(ASTCastExpression castExpr, Object data) {
             // the object will not implement SubItf anymore.
         } else if (isCastUnnecessary(castExpr, context, coercionType, operandType)) {
             reportCast(castExpr, data);
+        } else if (castExpr.getParent() instanceof ASTMethodCall
```

Comment:
You could use the `context` object for this: just verify it is an invocation context, like on line 91. The advantage is that it will match in any invocation context, not just method calls (this applies to constructor calls, including eg enum constructor calls).

Reply:
In this PR I did not cover the case where method *arguments* are unnecessarily cast, only the case where unnecessary cast happens for the method receiver. In the case covered here the context is `MISSING` rather than `INVOCATION`. But maybe we can handle both in the same PR?